### PR TITLE
Adds support for USE_GIT_URLS environment variable

### PR DESCRIPTION
**Pull request by Svenito**
Scripts clone via https by default, but by setting USE_GIT_URLS, all clones are done over git protocol

In my case I cannot clone over https due to network restrictions, which causes the `sys/install.sh` to hang. This should allow users to select a protocol that works. I imagine https is more commonly allowed, so it's the default.

### CODE REVIEW

**Comment by Svenito** [comment]
Are these due to a network timeout?

**Comment by radare** [comment]
Do not dupe the GIT_PREFIX. just put that in an .mk and include it on every makefile that needs it

**Comment by radare** [comment]
Which kind of network problems happening with https? it's because of a proxy? there are config options for git that can be globally setup to use a proxy with git.

**Comment by Svenito** [comment]
Thanks. I'll investigate and make the changes

**Comment by radare** [comment]
Yep. seems to fail because of a timeout.

**Comment by Svenito** [comment]
Ok, made the mk changes and travis passes. Can't determine what specifically is causing Appveyor to fail though. 

What specifically about the shell scripts needs to be changed, the way the env var is tested, or the replication of the test across the scripts?

**Comment by radare** [comment]
Appveyor is failing even without your changes . Dont worry about that

File: sys/vala.sh [hunk 1]
```diff
@@ -37,7 +37,11 @@ if [ -d vala ]; then
 	#sudo make uninstall
 	git pull
 else
-	git clone git://git.gnome.org/vala
+    if [ -z "$USE_GIT_URLS" ]; then
+        git clone https://git.gnome.org/vala
```
Comment (crowell) [inline sys/vala.sh]: this endpoint doesn't exist 

```
impactsite /tmp » git clone https://git.gnome.org/vala                                                                               130 ↵
Cloning into 'vala'...
fatal: repository 'https://git.gnome.org/vala/' not found
```

**Comment by crowell** [comment]
don't merge (yet), the vala repo isn't accessible over `https`

File: shlr/Makefile [hunk 2]
```diff
@@ -109,7 +110,7 @@ HFILES+=elf.h libtcc.h config.h i386-tok.h
 .PHONY: sdb-sync sync-sdb sdbclean
 tcc-sync sync-tcc:
 	rm -rf _
-	git clone --depth 1 git://repo.or.cz/tinycc.git _
+	git clone --depth 1 $(GIT_PREFIX)repo.or.cz/tinycc.git _
```
Comment (crowell) [inline shlr/Makefile]:
```
impactsite /tmp » git clone https://repo.or.cz/tinycc.git                                                                            130 ↵
Cloning into 'tinycc'...
fatal: unable to access 'https://repo.or.cz/tinycc.git/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
```

**Comment by radare** [comment]
Dont worry about merging. Its holidays here and ill be a bit afk tonight. I'll test properly before merging. I still dont understand why https is problematic

**Comment by radare** [comment]
why https is a problem? also, if he just wants to build r2 he doesn't needs any git pull or clone at all from any of those repositories. maybe only capstone, but the rest are supposed to be updated by me, in order to sync with stable/known versions. so, sys/install.sh will check if capstone is installed system wide and use it instead of pulling and building the capstone from git

**Comment by radare** [comment]
ping

**Comment by Svenito** [comment]
Sorry... Certain network rules don't allow git transactions over https, so I have to use git:// This is a problem when doing a `sys/install.sh` to get the latest git version, as the capstone repo is specified as https. You're right though, in most cases that will be the only thing to care about.

I am happy to edit this locally and carry on, but thought it might be nice to have the choice too, but I can see it has bigger knock-on effects.

**Comment by radare** [comment]
Then we should just handle this for capstone, not the rest of git urls. Right? Can you redo the patch acording to this?

**Comment by Svenito** [comment]
Sounds good. I'll get on it.

**Comment by radare** [comment]
Squashed and merged! thanks

**Pull request closed without merging**